pyasn1/codec/ber/decoder.py

```diff
@@ -138,27 +173,26 @@ def valueDecoder(self, substrate, asn1Spec,
                      tagSet=None, length=None, state=None,
                      decodeFun=None, substrateFun=None,
                      **options):
-        head, tail = substrate[:length], substrate[length:]
 
         if substrateFun:
             return substrateFun(self._createComponent(
                 asn1Spec, tagSet, noValue, **options), substrate, length)
 
-        if not head:
+        if isEmpty(substrate) or not length:
             raise error.PyAsn1Error('Empty BIT STRING substrate')
 
         if tagSet[0].tagFormat == tag.tagFormatSimple:  # XXX what tag to check?
 
-            trailingBits = oct2int(head[0])
+            trailingBits = ord(substrate.read(1))
```

Comment:
The feature of `oct2int` is that it isolates py2/py3 differences.

Reply:
There is a difference with string item selection. In case of read, you always get a bytes object and hence the ord.